pkg/template/api/validation/validation.go

```diff
@@ -51,3 +55,116 @@ func validateTemplateBody(template *api.Template) (allErrs field.ErrorList) {
+		for _, msg := range oapi.GetNameValidationFunc(validation.ValidateSecretName)(ti.Spec.SecretRef.Name, false) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.secretRef.name"), ti.Spec.SecretRef.Name, msg))
+		}
+	}
+	return
+}
+
+// ValidateTemplateBrokerUUIDMap tests if required fields in the TemplateBrokerUUIDMap are set.
+func ValidateTemplateBrokerUUIDMap(uuidmap *api.TemplateBrokerUUIDMap) (allErrs field.ErrorList) {
+	allErrs = validation.ValidateObjectMeta(&uuidmap.ObjectMeta, false, nameIsUUID, field.NewPath("metadata"))
+	allErrs = append(allErrs, validateTemplateInstanceRef(&uuidmap.Spec.TemplateInstanceRef, field.NewPath("spec.templateInstanceRef"))...)
+	for _, id := range uuidmap.Spec.BindingIDs {
+		for _, msg := range nameIsUUID(id, false) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.bindingIDs"), id, msg))
+		}
+	}
+	return
+}
+
+// ValidateTemplateBrokerUUIDMapUpdate tests if required fields in the TemplateBrokerUUIDMap are set during an update
+func ValidateTemplateBrokerUUIDMapUpdate(uuidmap, oldUuidmap *api.TemplateBrokerUUIDMap) (allErrs field.ErrorList) {
+	allErrs = validation.ValidateObjectMetaUpdate(&uuidmap.ObjectMeta, &oldUuidmap.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, validateTemplateInstanceRef(&uuidmap.Spec.TemplateInstanceRef, field.NewPath("spec.templateInstanceRef"))...)
+	for _, id := range uuidmap.Spec.BindingIDs {
+		for _, msg := range nameIsUUID(id, false) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec.bindingIDs"), id, msg))
+		}
+	}
+	return
+}
+
+var uuidRegex = regexp.MustCompile("^(?i)[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
+
+func nameIsUUID(name string, prefix bool) []string {
+	if uuidRegex.MatchString(name) {
+		return nil
+	}
+	return []string{"is not a valid UUID"}
+}
+
+func validateTemplateInstanceRef(ref *kapi.ObjectReference, fldPath *field.Path) (allErrs field.ErrorList) {
+	if len(ref.Kind) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("kind"), ""))
+	} else if ref.Kind != "TemplateInstance" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("kind"), ref.Kind, "must be TemplateInstance"))
+	}
+
+	if len(ref.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
+	} else {
+		for _, msg := range oapi.GetNameValidationFunc(validation.ValidatePodName)(ref.Name, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("name"), ref.Name, msg))
+		}
+	}
+
+	if len(ref.Namespace) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("namespace"), ""))
+	} else {
+		for _, msg := range oapi.GetNameValidationFunc(validation.ValidateNamespaceName)(ref.Namespace, false) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("namespace"), ref.Namespace, msg))
+		}
+	}
+
+	return allErrs
+}
+
+func validateTemplateInstanceImpersonation(oc *client.Client, ctx kapi.Context, ti *api.TemplateInstance) field.ErrorList {
```

Comment:
> @deads2k this is where I'm trying to confirm that the current user has the right to impersonate ti.Spec.ImpersonateUser.

Validation should not rely on anything outside of the object itself.  Otherwise you can get inconsistent results depends on when or where you run it.